src/auth/auth.service.ts

```diff
@@ -45,8 +53,9 @@ export class AuthServices {
             firstName,
             lastName
         }
-        this.userRepository.save(saveData);
-        const payload = { firstName, lastName, email };
+        const data = await this.userRepository.save(saveData);
+        console.log(data);
+        const payload = { firstName, lastName, email, roleId: data.roleId };
```

Comment:
please see [this article](https://docs.nestjs.com/techniques/serialization) and use `class-transformer` for serialization

Reply:
same for login